Data/DataModificator.cs

```diff
@@ -18,26 +18,26 @@ public class DataModificator : IDataModificator
 
         public async Task AddCharityEvent(IGenericKarmaItem newCharityEvent)
         {
-            List<IGenericKarmaItem> charityEventList = (await m_fileContentProvider.ReadFromFileAsync()).ToList();
+            List<IGenericKarmaItem> charityEventList = (await m_fileContentProvider.ReadItemFromFileAsync()).ToList();
             charityEventList.Add(newCharityEvent);
-            await m_fileContentProvider.WriteToFileAsync(charityEventList);
+            await m_fileContentProvider.WriteItemToFileAsync(charityEventList);
         }
 
         public async Task RemoveCharityEvent(Guid charityEventId)
         {
-            List<IGenericKarmaItem> charityEventList = (await m_fileContentProvider.ReadFromFileAsync()).ToList();
+            List<IGenericKarmaItem> charityEventList = (await m_fileContentProvider.ReadItemFromFileAsync()).ToList();
             IEnumerable<IGenericKarmaItem> newCharityList = charityEventList.Where(p => p.Id != charityEventId);
-            await m_fileContentProvider.WriteToFileAsync(newCharityList);
+            await m_fileContentProvider.WriteItemToFileAsync(newCharityList);
```

Comment:
Incorrect name. WriteItemToFile implies that it will write 1 item, but you pass a list

Reply:
Name will be removed after the last fix,